classes/log/store.php

```diff
@@ -89,6 +89,7 @@ protected function insert_event_entries(array $events) {
     public function process_events(array $events) {
         global $DB;
         global $CFG;
+        require_once(__DIR__ . '/../../version.php');
```

Comment:
Any idea why that works that way but not the other? We have the plugin on a local Moodle 3.5 instance using the other way.

Reply:
Great question, I'm running 7.0.6 on my local VM and it was throwing errors glore over that. 

Just like the tenery functions for names. ````$site_name = $site->fullname ? $site->fullname : 'A Moodle site';```` if you just leave it as ````$site_name = $site->fullname ?: 'A Moodle site';```` it returns null. Not sure why that is happening, from my understanding that later code line should return $fullname not null.

Reply:
Actually, my VM is running windows. What are your dev machines running? There is a chance this is an OS specific issue. In which case, how do you want to handle it? Because this looks like this should work either way, the way I have it now. Windows may have a more strict bug checking because I have xdebug on the windows server.

Reply:
Not sure about the ternary's, don't mind changing those ternary's to stop using `?:` that's fine. Just not sure why the `require_once` is causing a problem for that particular file, but not the other one above it `require_once(__DIR__ . '/../../src/autoload.php');`.